eng/WpfArcadeSdk/tools/InjectModuleInitializer.targets

```diff
@@ -62,8 +62,8 @@
           try
           {
             var bitness = (DotNetFrameworkArchitecture)Enum.Parse(typeof(DotNetFrameworkArchitecture), DotNetBitness);
-            ILAsm = ToolLocationHelper.GetPathToDotNetFrameworkFile("ILAsm.exe", TargetDotNetFrameworkVersion.Latest, bitness);
-            ILDAsm = ToolLocationHelper.GetPathToDotNetFrameworkSdkFile("ILDAsm.exe", TargetDotNetFrameworkVersion.Latest, bitness);
+            ILAsm = ToolLocationHelper.GetPathToDotNetFrameworkFile("ILAsm.exe", TargetDotNetFrameworkVersion.VersionLatest, bitness);
+            ILDAsm = ToolLocationHelper.GetPathToDotNetFrameworkSdkFile("ILDAsm.exe", TargetDotNetFrameworkVersion.VersionLatest, bitness);
```

Comment:
This reverts the fix for #5059, hence re-introduces that build break.  Which was incomprehensible to me and everyone else for many days.

If you must do this, is there somewhere you can announce that building release/3.1 requires having installed the .NET Framework 4.6.2 developer pack?  In a way that's discoverable by users who hit the build break.

It's also disturbing that the testing for #5059 didn't discover the problem.   There's a gap that needs filling.

Reply:
Yes, we could add it to the requirements in the documentation and update the .vsconfig (for 3.1).

Reply:
@SamBent See my comment for the reason it didn't break the build: https://github.com/dotnet/wpf/issues/5375#issuecomment-926116560.

In short, ILAsm fails but the Task ILAsm returns true whether or not the process exited with 0 or not. Since the ILAsm simply overrides the original dll, it doesn't break the build and the original dll is kept in the output. One quick fix would be to replace [this](https://github.com/dotnet/wpf/blob/v3.1.19/eng/WpfArcadeSdk/tools/InjectModuleInitializer/ILAsmTask.cs#L291-L292) by this:
```
Process process = Process.Start(startInfo);
process.WaitForExit();
return process.ExitCode == 0;
```

Reply:
I put the documentation update in a separate PR: https://github.com/dotnet/wpf/pull/5402.